### PR TITLE
mgr/status: ceph osd status support json format

### PR DESCRIPTION
**Pull request by shminjs**
ceph osd status -f json/json-pretty return string format, add
json/json-pretty support

Fixes: https://tracker.ceph.com/issues/52713
Signed-off-by: shimin <shminjs@outlook.com>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
I think this needs a test

**Comment by shminjs** [comment]
> I think this needs a test

I add an test

**Comment by shminjs** [comment]
> I think this needs a test

Updated. I add a test for this command.

**Review by pereman2 — APPROVED** [review]
worked fine locally

File: src/pybind/mgr/status/module.py [hunk 1]
```diff
@@ -279,7 +279,7 @@ def handle_fs_status(self,
             return HandleCommandResult(stdout=output)
 
     @CLIReadCommand("osd status")
-    def handle_osd_status(self, bucket: Optional[str] = None) -> Tuple[int, str, str]:
+    def handle_osd_status(self, bucket: Optional[str] = None, format: str = 'plain') -> Tuple[int, str, str]:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/status/module.py:282]: Please have a look at 

https://github.com/ceph/ceph/blob/7af7255fef5d054e94fa308eff4d884967cab798/src/pybind/mgr/orchestrator/module.py#L372

to properly propagate possible values to the CLI tool

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/status/module.py (hunk 1 above)
Comment (shminjs) [reply src/pybind/mgr/status/module.py]: Yes, already change to this format. when use xml, it raise a ValueError: 'xml' is not a valid Format

**Review by sebastian-philipp — APPROVED** [review]

File: src/pybind/mgr/status/module.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/status/module.py]: yes, cephadm is also affected by this. But as XML is nowadays less common, I'm ok with ignoring this for now.

**Comment by shminjs** [comment]
jenkins make check

**Comment by shminjs** [comment]
jenkins test make check

**Comment by shminjs** [comment]
jenkins test api

**Review by joscollin — COMMENTED** [review]
Tested this fix. Now it started showing this error for the `Format`s that you haven't considered from the usage/help.

```
Error EINVAL: Traceback (most recent call last):
  File "/home/jcollin/test_perf/ceph/ceph/src/pybind/mgr/mgr_module.py", line 1591, in _handle_command
    return CLICommand.COMMANDS[cmd['prefix']].call(self, cmd, inbuf)
  File "/home/jcollin/test_perf/ceph/ceph/src/pybind/mgr/mgr_module.py", line 411, in call
    kwargs = self._collect_args_by_argspec(cmd_dict)
  File "/home/jcollin/test_perf/ceph/ceph/src/pybind/mgr/mgr_module.py", line 404, in _collect_args_by_argspec
    kwargs[k] = CephArgtype.cast_to(tp, v)
  File "/home/jcollin/test_perf/ceph/ceph/src/pybind/ceph_argparse.py", line 242, in cast_to
    return tp(v)
  File "/usr/lib/python3.6/enum.py", line 293, in __call__
    return cls.__new__(cls, value)
  File "/usr/lib/python3.6/enum.py", line 535, in __new__
    return cls._missing_(value)
  File "/usr/lib/python3.6/enum.py", line 548, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 'xml-pretty' is not a valid Format
```

**Comment by sebastian-philipp** [comment]
> Tested this fix. Now it started showing this error for the `Format`s that you haven't considered from the usage/help.
> 
> ```
> ...snip...
>   File "/usr/lib/python3.6/enum.py", line 548, in _missing_
>     raise ValueError("%r is not a valid %s" % (value, cls.__name__))
> ValueError: 'xml-pretty' is not a valid Format
> ```

In my opinion, we should look into why the ceph binary is forwarding `xml-pretty` to the command hander despite it not being marked as a valid enum. mgr/orchestrator is affected by this bug in the ceph binary as well.

File: qa/tasks/mgr/test_status.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def setUp(self):
+        super(TestStatus, self).setUp()
+        self.setup_mgrs()
+
+    def test_osd_status(self):
+        """
+        That `ceph osd status` command functions.
+        """
+
+        s = self.mgr_cluster.mon_manager.raw_cluster_cmd("osd", "status")
```
Comment (joscollin) [inline qa/tasks/mgr/test_status.py:21]: This test cannot find `ceph osd status` command.

```
teuthology.exceptions.CommandFailedError: Command failed with status 22: ['./bin/ceph', 'osd', 'status']
```

**Review by joscollin — CHANGES_REQUESTED** [review]
Please fix the above mentioned issues.

**Pull request closed without merging**